### ISSUE TITLE
Add copy fallback for reflink-based linker

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/uv#1444]
See: https://twitter.com/mpr0010/status/1758299951254388875.

**Comment by humanzz** [comment]
Same issue here, and it's on a mac.

I have my python package in case-sensitive volume e.g. `/Volumnes/casesensitive` and when I try to `uv pip install -e '.[dev,testing]'`, I get an errors along the lines of
```
error: Failed to build editables
  Caused by: Failed to build editable: file:///Volumes/casesensitive/MyTestPackage
  Caused by: Failed to install requirements from build-system.requires (install)
  Caused by: Failed to install build dependencies
  Caused by: Failed to install: setuptools-69.1.0-py3-none-any.whl (setuptools==69.1.0)
  Caused by: Failed to reflink /Volumes/casesensitive/uvc/archive-v0/Zgihw_1YQkIZZNdz81Rtf/setuptools-69.1.0.dist-info to /private/var/folders/43/29_b2rv50rj6c9t29s0t5w5whc959c/T/.tmpYrzRHM/.venv/lib/python3.10/site-packages/setuptools-69.1.0.dist-info
  Caused by: Cross-device link (os error 18)
```

**Comment by charliermarsh** [comment]
Thanks! This one should be straightforward to fix. (In the interim, I might suggest setting `UV_CACHE_DIR` to something in the same volume as your projects, it will also speed up installation.)

**Comment by humanzz** [comment]
No worries, I'm still testing this out. I had tried `UV_CACHE_DIR` to be on the same volume as my projects, but as can be seen in the earlier logs, there's a `/private/var...` path that popped up, that I don't think I have control over.

**Comment by humanzz** [comment]
I looked again into it, and tried out with the newer `0.1.4` version.

Setting the cache dir to be on the same volume
1. Succeeds if I install normally `uv pip install MyTestPackage[dev,testing]`
2. Fails if I install in editable mode  `uv pip install -e '.[dev,testing]'` and the failure shows `/private/var/...`.

**Comment by charliermarsh** [comment]
And in (2), it fails with the cross-device link error seen above?

**Comment by humanzz** [comment]
Yes. I edited the message to add more verbose output

**Comment by zanieb** [comment]
Hm so in `pip install` we create a temporary directory in the current virtual environment

https://github.com/astral-sh/uv/blob/b5617198f35c89a33f073c677ed9164bb661509c/crates/uv/src/commands/pip_install.rs#L188

https://github.com/astral-sh/uv/blob/b5617198f35c89a33f073c677ed9164bb661509c/crates/uv/src/commands/pip_install.rs#L360

but eventually we get to

https://github.com/astral-sh/uv/blob/12fea1d058083a4f8a032ca8295a7d95a776292e/crates/uv-build/src/lib.rs#L286

via 

https://github.com/astral-sh/uv/blob/2586f655bbf650a9797c8f88b6d9066eefe7a3dc/crates/uv-dispatch/src/lib.rs#L274

https://github.com/astral-sh/uv/blob/340cb67a8b2a708996bd87369d2cadf48ef9df63/crates/uv-distribution/src/source/mod.rs#L954

where we create a temporary directory to actually perform the build but it is not scoped to the current virtual environment or our cache (hence your issue)

**Comment by humanzz** [comment]
I've just tried the `0.1.5` release and it finally works :)

I haven't said it earlier in my comments here, and on other issues, but I have to say it now, Thank you for your awesome work... this is great ⚡ 🚀

**Comment by ifplusor** [comment]
> I've just tried the `0.1.5` release and it finally works :)
> 
> I haven't said it earlier in my comments here, and on other issues, but I have to say it now, Thank you for your awesome work... this is great ⚡ 🚀

I run `uv` in an additional APFS (Case-sensitive) volume, but it does not work.

**Comment by zanieb** [comment]
Thank you!

@ifplusor we still need to add a fallback copy as described in this issue, but until then you should be able to set `UV_CACHE_DIR` to avoid copying across volumes.

**Comment by charliermarsh** [comment]
Should be fixed in v0.1.7 (out now).
